pkg/webhook/validation/uservalidation.go

```diff
@@ -0,0 +1,20 @@
+package validation
+
+import (
+	"k8s.io/utils/strings/slices"
+)
+
+const (
+	authenticatedGroup = "system:authenticated"
+	mastersGroup       = "system:masters"
+)
+
+// TODO: Get valid user names as flag and check to validate those user names.
+
+// ValidateUserGroups checks to see if user is authorized to make a request to the hub cluster's api-server.
+func ValidateUserGroups(groups []string) bool {
```

Comment:
This is a tricky one; As Arvind you have pointed out in our previous sync, the `authenticated` group might not be a good choice. Let's discuss this further if it is OK with you.

Reply:
We only allow user from system:masters group to make changes to fleet CRDs